core/src/main/java/nl/nn/adapterframework/pipes/XmlIf.java

```diff
@@ -116,35 +126,25 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 			}
 		}
 
-		// log.debug(getLogPrefix(session) + "input value is [" + sInput + "]");
-		
 		if (tp!=null) {
 			try {
-				forward = tp.transform(sInput,null, isNamespaceAware());
+				Map<String,Object> parametervalues = null;
+				ParameterList parameterList = getParameterList();
+				if (!parameterList.isEmpty()) {
+					message.preserve();
```

Comment:
`message.preserve()` should only be done if `parameterList.requiresInputValueForResolution()`